jena-fuseki2/jena-fuseki-docker/Dockerfile

```diff
@@ -0,0 +1,111 @@
+## Licensed to the Apache Software Foundation (ASF) under one or more
+## contributor license agreements.  See the NOTICE file distributed with
+## this work for additional information regarding copyright ownership.
+## The ASF licenses this file to You under the Apache License, Version 2.0
+## (the "License"); you may not use this file except in compliance with
+## the License.  You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+## 
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+## Apache Jena Fuseki server Dockerfile.
+
+## This Dockefile builds a reduced footprint container.
+
+ARG OPENJDK_VERSION=14
+ARG ALPINE_VERSION=3.12.0
+ARG JENA_VERSION=""
+
+# Internal, passed between stages.
+ARG FUSEKI_DIR=/fuseki
+ARG FUSEKI_JAR=jena-fuseki-server-${JENA_VERSION}.jar
+ARG JAVA_MINIMAL=/opt/java-minimal
+
+## ---- Stage: Download and build java.
+FROM openjdk:${OPENJDK_VERSION}-alpine AS base
+
+ARG JAVA_MINIMAL
+ARG JENA_VERSION
+ARG FUSEKI_DIR
+ARG FUSEKI_JAR
+ARG REPO=https://repo1.maven.org/maven2
+ARG JAR_URL=${REPO}/org/apache/jena/jena-fuseki-server/${JENA_VERSION}/${FUSEKI_JAR}
+
+RUN [ "${JENA_VERSION}" != "" ] || { echo -e '\n**** Set JENA_VERSION ****\n' ; exit 1 ; }
+RUN echo && echo "==== Docker build for Apache Jena Fuseki ${JENA_VERSION} ====" && echo
+
+# Alpine: For objcopy used in jlink
+RUN apk add --no-cache curl binutils
+
+## -- Fuseki installed and runs in /fuseki.
+WORKDIR $FUSEKI_DIR
+
+## -- Download the jar file.
+COPY download.sh .
+RUN chmod a+x download.sh
+
+RUN ./download.sh --chksum sha1 "$JAR_URL"
```

Comment:
Although I think docker doesn't provide checksum verification, so it would justify having the shell script.

Reply:
My experience is that ADD runs each time, presumably because it does not know whether the file at the URL has changed. If it was a local filesystem file, it has more to go on, e..g file modification time, but for a URL that information isn't there.

Maybe I was using it wrongly; I didn't chase it further because you are spot-on that it does not check the checksum, and with large files from maven.central I had a partial download during a docker build.

At £job, we run the process from `make`.  It is the `Makefile` that downloads and caches the file then docker `COPY`s from local filesystem into the docker container build. But that assumes a `make` setup so I put it the steps in the build stage of the Dockerfile.

All that said, I don't use docker that much so advice on better practice is very welcome, especially if attached to a PR!

Reply:
Thanks for the explanation @afs ! Let's go with this approach for it works and for the checksum. We can change it later if necessary :tada: :rocket: